### PR TITLE
Correctly decode tszip cols

### LINKED ISSUE #190: edges table viewer: incorrect right coordinate displayed?
Screenshot 2024-10-11 095758

image

There seems to be a difference between the right coordinates displayed in the Edge table viewer vs the tree sequence edges table.

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #190
Fixed other odd columns too and added tests for all columns.

### CODE REVIEW

File: tsbrowse/model.py [hunk 1]
```diff
@@ -54,20 +64,34 @@ def __init__(self, tsbrowse_path):
             array_names -= set(ragged_array_names)
             array_names -= {"metadata_schema"}
             array_names -= {f"{name}_offset" for name in ragged_array_names}
-            arrays = {name: root[table_name][name][:] for name in array_names}
-            ragged_array_names -= {"metadata"}
+            arrays = {}
```
Comment (jeromekelleher) [inline tsbrowse/model.py:67]: This feels like we're reproducing tszip code to me - why do we need to do this?

**Review by jeromekelleher — COMMENTED** [review]
I don't understand why we need to do this tbh - shouldn't we let tszip handle this? If we do
```
ts = tszip.load(tsbrowse_path)
root = zarr.open(tsbrowse_path)
# e.g.
edges_df = {
     "left": ts.edges_left,
     "right": ts.edges_right, 
      ...
 } 
```
then we let tszip deal with these details, and we don't become depending on an implementation detail of tszip?

File: tsbrowse/model.py (hunk 1 above)
Comment (benjeffery) [reply tsbrowse/model.py:67]: I guess your suggesting that for a fixed list of `table/col` we use tskit, and the rest load using zarr? Yes, would probably be cleaner.

File: tsbrowse/model.py [hunk 2]
```diff
@@ -54,44 +43,35 @@ def __init__(self, tsbrowse_path):
             "migrations",
             "provenances",
         ]:
+            ts_table = getattr(self.ts.tables, table_name)
```
Comment (jeromekelleher) [inline tsbrowse/model.py:46]: That's generating a fresh copy of the tables each time. Get a reference to the tables before the loop body

**Review by jeromekelleher — COMMENTED** [review]
Minor perf issue

File: tsbrowse/model.py (hunk 2 above)
Comment (benjeffery) [reply tsbrowse/model.py]: Fixed!

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**